### ISSUE TITLE
Official stance on using newer Terraform versions

### ISSUE THREAD

**Issue by jeanduplessis** [upbound/provider-terraform#287]
This issue documents the official stance that we will not adopt any Terraform version in this provider released under the BSL license. 

As such, this provider will be frozen on Terraform 1.5.7. 

We are also not open to allowing end-users to specify the version of Terraform they want to use using a solution like `tfenv`.

**Comment by jeanduplessis** [comment]
@senare The decision is the culmination of the legal opinion sought on the license terms of Terraform and upcoming guidelines from the CNCF: https://github.com/cncf/foundation/pull/843/files

While we appreciate the desire to see one unified solution, this provider will not look to provide OpenTofu or Terragrunt integration, and if the community wants to pursue an OpenTofu solution, it should be created as a separate provider.

**Comment by RickardCardell** [comment]
Correct me if I'm wrong, but the provider for OpenTofu seems to exist now. Adding a link here for sake of visibility: https://github.com/upbound/provider-opentofu

**Comment by jeanduplessis** [comment]
@RickardCardell correct. Any OpenTofu support will happen in that separate provider, not this one.

**Comment by jeanduplessis** [comment]
⚠️ Please note that this repo has migrated to https://github.com/crossplane-contrib/provider-terraform 

Due to GitHub's limitations on transferring issues across organizations, this issue will be closed. 
Please re-open your issue upstream if you still want it considered.

Note that Upbound will no longer actively invest in this provider, but it will endeavor to continue supporting community contributions and publishing new releases as needed.

For users interested in using newer Terraform versions, please note that provider-terraform will NOT support versions of Terraform under the BUSL license (anything after 1.5.x). We encourage users who need newer capabilities to consider https://github.com/upbound/provider-opentofu.

**Comment by maxsxu** [comment]
Hi, @jeanduplessis thanks for this info.

It'd better to archive this repo and update the depreciation notice.

I've created a [PR](https://github.com/upbound/provider-terraform/pull/427) to update the depreciation notice.
